### PR TITLE
Automatic Versioning for Releases

### PR DESCRIPTION
**Pull request by hillmich**
- Get version from git tag
- Drop version in CMakeLists.txt
- Correct version for RtD
- Streamline python build config

### CODE REVIEW

File: mqt/ddsim/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 pybind11_add_module(py${PROJECT_NAME} bindings.cpp)
 target_link_libraries(py${PROJECT_NAME} PUBLIC ${PROJECT_NAME} MQT::qfr_python pybind11_json)
+target_compile_definitions(py${PROJECT_NAME} PRIVATE VERSION_INFO=${DDSIM_VERSION_INFO})
```
Comment (burgholzer) [inline mqt/ddsim/CMakeLists.txt:3]: there's a newline missing at the end of this file.

File: mqt/ddsim/bindings.cpp [hunk 2]
```diff
@@ -10,13 +10,12 @@
 #include "qiskit/QasmQobjExperiment.hpp"
 #include "qiskit/QuantumCircuit.hpp"
 
-#include "pybind11/complex.h"
 #include <pybind11/numpy.h>
```
Comment (burgholzer) [inline mqt/ddsim/bindings.cpp:13]:
```suggestion
#include "pybind11/numpy.h"
```

**Review by burgholzer — APPROVED** [review]
Couple of really small remarks, but otherwise this LGTM.

File: mqt/ddsim/bindings.cpp (hunk 2 above)
Comment (hillmich) [reply mqt/ddsim/bindings.cpp:13]: As disscussed, I'll change the `"` to `<>`.

**Pull request merged**